cmd/ig/main.go

```diff
@@ -79,7 +76,38 @@ func main() {
 	rootCmd.AddCommand(common.NewLoginCmd())
 	rootCmd.AddCommand(common.NewLogoutCmd())
 
+	// evaluate flags early; this will make sure that flags for host are evaluated before
+	// calling host.Init()
+	err := commonutils.ParseEarlyFlags(rootCmd, os.Args[1:])
```

Comment:
`host.Init()` is called in `RunE` which would be executed in `rootCmd.Execute`, therefore, calling this function after adding all sub-commands so that flags are parsed (solves the above issue) but before `rootCmd.Execute` which satisfies the comment condition.

Reply:
Is this necessary for the use of env variables or it is just something you want to add to make the code more readable/organised?

Reply:
It solves the issue of `f.Changed` being false even if we set the flag values of sub-commands. 
Therefore, parsing flags after adding all sub-commands.